modules/document_repository/php/files.class.inc

```diff
@@ -0,0 +1,449 @@
+     * @throws \DatabaseException
+     *
+     * @return void
+     */
+    function deleteFile($rid): void
+    {
+        // create Database object
+        $DB       = \Database::singleton();
+        $user     = \User::singleton();
+        $Notifier = new \NDB_Notifier(
+            "document_repository",
+            "delete"
+        );
+
+        $fileName = $DB->pselectOne(
+            "SELECT File_name FROM document_repository
+             WHERE record_id =:identifier",
+            array(':identifier' => $rid)
+        );
+        $userName = $DB->pselectOne(
+            "SELECT uploaded_by FROM document_repository
+             WHERE record_id =:identifier",
+            array(':identifier' => $rid)
+        );
+        $dataDir  = $DB->pselectOne(
+            "SELECT Data_dir FROM document_repository WHERE record_id =:identifier",
+            array(':identifier' => $rid)
+        );
+
+        //if user has document repository delete permission
+        if ($user->hasPermission('document_repository_delete')) {
+            $DB->delete("document_repository", array("record_id" => $rid));
+            $msg_data['deleteDocument'] = $baseURL. "/document_repository/";
+            $msg_data['document']       = $fileName;
+
+            $Notifier->notify($msg_data);
+        }
+
+        $path = __DIR__ . "/../user_uploads/$dataDir";
+
+        if (file_exists($path)) {
+            unlink($path);
+        }
+
+    }
+    /**
+     * Returns a list of categories fields from database
+     *
+     * @param String $id id of the Doc file
+     *
+     * @return array
+     */
+    function getUploadDocFields(string $id): array
+    {
+        $db    = \Database::singleton();
+        $query = $db->pselect(
+            "SELECT * FROM document_repository_categories",
+            array()
+        );
+        //categories
+        $categoriesList = array();
+        foreach ($query as $value) {
+             $arr = $this->parseCategory($value);
+             $categoriesList[$arr['id']] =$arr['name'];
+        }
+        //site
+        $siteList = \Utility::getSiteList(false);
+        //instrument
+        $instruments     = $db->pselect(
+            "SELECT Test_name FROM test_names ORDER BY Test_name",
+            array()
+        );
+        $instrumentsList = $this->toDocSelect($instruments, "Test_name", null);
+        //docFile
+        $docData = null;
+        if (isset($id)) {
+            $idDocFile = $id;
+            $docData   = $db->pselectRow(
+                "SELECT " .
+                "record_id as id, " .
+                "PSCID as pscid, " .
+                "File_category as category," .
+                "visitLabel, " .
+                "Instrument as instrument, " .
+                "For_site as forSite, " .
+                "comments, " .
+                "File_Name as fileName, " .
+                "version " .
+                "FROM document_repository " .
+                " WHERE record_id = $idDocFile",
```

Comment:
I didn't trace where the `$id` parameter is passed from, but this appears to be vulnerable to SQL injection. The value should be passed as a prepared statement argument.